agent/engine/serviceconnect/manager_linux.go

```diff
@@ -402,3 +414,16 @@ func getECSAgentLogPathContainer() string {
 	}
 	return path.Dir(ecsLogFilePath)
 }
+
+// GetSupportedAppnetInterfaceVerToCapabilities returns service connect capabilities
+// supported by ECS Agent to register for a selected AppNet version.
+// Suppose we decide to register ecs.service-connect.v2 capability for new AppNet version (ex: 1.24.0.0),
+// now if ecs.service-connect.*v1* is continuously being supported by 1.24.0.0,
+// we will then register multiple capabilities.
+// {
+//    "v1": ["ecs.service-connect.v1"],
+//    "v2": ["ecs.service-connect.v1", "ecs.service-connect.v2"]
+// }
+func (agent *manager) GetSupportedAppnetInterfaceVerToCapabilities() []string {
+	return supportedAppnetInterfaceVerToCapability[agent.AppnetInterfaceVersion]
```

Comment:
I feel that it may better to have caller (`appendServiceConnectCapabilities`) explicitly pass in the appnet version. Currently it first calls loader, loader will set the global appnet version, which is used by this method, so we are introducing some dependencies between those two methods which couldn't been independent. 

Let's maybe have below in `appendServiceConnectCapabilities` instead:
```
loadedVer, err := agent.LoadImage(...)
// handle err
capabilities := agent.GetCapabilitiesForAppnetInterfaceVersion(loadedVer)
// if nil, throw warning
// append the capabilities
```
Overall feels a bit more clear to me. What do you think?

Reply:
The LoadImage() function is used to load pause container image as well, thought returning version might not add much value in that context. 
I have added a get and set functions to manage the appnet version, hope it makes it a bit clear, please let me know:)